app/models/user.rb

```diff
@@ -738,8 +738,13 @@ def self.system_avatar_template(username)
 
   def self.letter_avatar_color(username)
     username ||= ""
-    color = LetterAvatar::COLORS[Digest::MD5.hexdigest(username)[0...15].to_i(16) % LetterAvatar::COLORS.length]
-    color.map { |c| c.to_s(16).rjust(2, '0') }.join
+    if SiteSetting.restrict_letter_avatar_colors.present?
+      colors = SiteSetting.restrict_letter_avatar_colors.split("|")
+      color = colors[Digest::MD5.hexdigest(username)[0...15].to_i(16) % colors.length]
+    else  
+      color = LetterAvatar::COLORS[Digest::MD5.hexdigest(username)[0...15].to_i(16) % LetterAvatar::COLORS.length]
```

Comment:
`Digest::MD5.hexdigest(username)[0...15].to_i(16) % LetterAvatar::COLORS.length`

This line of logic is being duplicated which means we have two places to update if the logic should change in the future. I think it'll be better if we can refactor this into a method of its own so that we won't ever have a case where the logic will differ for the two different code paths.